src/users/userController.js

```diff
@@ -0,0 +1,172 @@
+const Joi = require("joi");
+const {
+  Types: { ObjectId },
+} = require("mongoose");
+const bcryptjs = require("bcryptjs");
+const jwt = require("jsonwebtoken");
+const userModel = require("./userModel");
+
+class UserController {
+  constructor() {
+    this._costFactor = 4;
+  }
+  get register() {
+    return this._register.bind(this);
+  }
+  validateRegisterAndLogin(req, res, next) {
+    const schema = Joi.object({
+      email: Joi.string().required(),
+      password: Joi.string().required(),
+      subscription: Joi.string(),
+    });
+
+    const validation = schema.validate(req.body);
+    if (validation.error) {
+      const error = new Error();
+      error.message = validation.error;
+      return res.status(400).send(error);
+    }
+
+    next();
+  }
+  validateUserId(req, res, next) {
+    const { userId } = req.params;
+
+    if (!ObjectId.isValid(userId)) {
+      return res.status(400).send();
+    }
+    next();
+  }
+  async _register(req, res, next) {
+    const { email, password, subscription } = req.body;
+    const isUserExist = await userModel.findOne({ email });
+
+    if (isUserExist) {
+      const error = new Error();
+      error.message = "Email in use";
+      return res.status(409).send(error);
+    }
+
+    const hashedPassword = await bcryptjs.hash(
```

Comment:
const hashedPassword = await bcryptjs.hash(
      password,
      this._costFactor
);